eslint.config.js

```diff
@@ -0,0 +1,249 @@
+ */
+
+import { configs, plugins } from 'eslint-config-airbnb-extended';
+import { rules as prettierConfigRules } from 'eslint-config-prettier';
+import prettierPlugin from 'eslint-plugin-prettier';
+import reactRefreshPlugin from 'eslint-plugin-react-refresh';
+import js from '@eslint/js';
+
+const jsConfig = [
+    // ESLint Recommended Rules
+    {
+        name: 'js/config',
+        ...js.configs.recommended,
+    },
+    // Stylistic Plugin
+    plugins.stylistic,
+    // Import X Plugin
+    plugins.importX,
+    // Airbnb Base Recommended Config
+    ...configs.base.recommended,
+];
+
+const reactConfig = [
+    // React Plugin
+    plugins.react,
+    // React Hooks Plugin
+    plugins.reactHooks,
+    // React JSX A11y Plugin
+    plugins.reactA11y,
+    // Airbnb React Recommended Config
+    ...configs.react.recommended,
+];
+
+const typescriptConfig = [
+    // TypeScript ESLint Plugin
+    plugins.typescriptEslint,
+    // Airbnb Base TypeScript Config
+    ...configs.base.typescript,
+    // Airbnb React TypeScript Config
+    ...configs.react.typescript,
+];
+
+const prettierConfig = [
+    // Prettier Plugin
+    {
+        name: 'prettier/plugin/config',
+        plugins: {
+            prettier: prettierPlugin,
+        },
+    },
+    // Prettier Config (disable conflicting rules)
+    {
+        name: 'prettier/config',
+        rules: {
+            ...prettierConfigRules,
+            'prettier/prettier': 'warn',
+        },
+    },
+];
+
+const projectConfig = [
+    {
+        name: 'project/ignores',
+        ignores: ['dist', 'build', 'coverage'],
+    },
+    {
+        name: 'project/settings',
+        settings: {
+            react: {
+                version: 'detect',
+            },
+            // TypeScript import resolver settings
+            'import-x/parsers': {
+                '@typescript-eslint/parser': ['.ts', '.tsx'],
+            },
+            'import-x/resolver': {
+                typescript: {
+                    alwaysTryTypes: true,
+                    project: './tsconfig.json',
+                },
+            },
+        },
+    },
+    // React Refresh Plugin
+    {
+        name: 'project/react-refresh',
+        files: ['**/*.{jsx,tsx}'],
+        plugins: {
+            'react-refresh': reactRefreshPlugin,
+        },
+        rules: {
+            'react-refresh/only-export-components': ['error', { allowConstantExport: true }],
+        },
+    },
+    // React JSX Runtime (prevents "React must be in scope" errors)
+    {
+        name: 'project/react-jsx-runtime',
+        files: ['**/*.{jsx,tsx}'],
+        rules: {
+            'react/react-in-jsx-scope': 'off',
+            'react/jsx-uses-react': 'off',
+        },
+    },
+    // Custom rules
+    {
+        name: 'project/rules',
+        rules: {
+            // Code style
```

Comment:
It's a lot of broken rules, if we can't use the same as before, we should really decide now which one we take or not...